parser_library/src/lsp/document_symbol_item.h

```diff
@@ -0,0 +1,41 @@
+
+
+#ifndef LSP_DOCUMENT_SYMBOL_ITEM_H
+#define LSP_DOCUMENT_SYMBOL_ITEM_H
+
+#include <algorithm>
+#include <string>
+#include <vector>
+
+#include "protocol.h"
+
+namespace hlasm_plugin::parser_library::lsp {
+
+using document_symbol_list_s = std::vector<document_symbol_item_s>;
+
+// representation of document symbol item based on LSP
+struct document_symbol_item_s
+{
+public:
+    document_symbol_item_s(sequence<char> name, document_symbol_kind kind, range symbol_range);
+    document_symbol_item_s(
+        sequence<char> name, document_symbol_kind kind, range symbol_range, document_symbol_list_s children);
+
+    // several features of document symbol item from LSP
+    sequence<char> name;
```

Comment:
This structure should own the response. The `document_symbol_item` then correctly provides view via sequence<char>
```suggestion
    document_symbol_item_s(std::string name, document_symbol_kind kind, range symbol_range);
    document_symbol_item_s(
        std::string name, document_symbol_kind kind, range symbol_range, document_symbol_list_s children);

    // several features of document symbol item from LSP
    std::string name;
```